File: non_native/lib/view/components/delete_bird_dialog.dart
import 'package:flutter/material.dart';

import '../../model/bird.dart';

class DeleteBirdDialog extends StatelessWidget {
  final bool showDialog;
  final Bird? birdToDelete;
  //final Function(int) onDeleteConfirmed;
  final Future<void> Function(int) onDeleteConfirmed;
  final Function onDismiss;

  const DeleteBirdDialog({
    super.key,
    required this.showDialog,
    required this.birdToDelete,
    required this.onDeleteConfirmed,
    required this.onDismiss,
  });

  @override
  Widget build(BuildContext context) {
    return showDialog
        ? AlertDialog(
            title: const Text("Delete Bird"),
            content:
                Text("Are you sure you want to delete ${birdToDelete?.name}?"),
            actions: [
              TextButton(
                onPressed: () => onDismiss(),
                child: const Text("Cancel"),
              ),
              TextButton(
                onPressed: () async {
                  await onDeleteConfirmed(birdToDelete!.id);
                },
                child: const Text("Confirm"),
              ),
            ],
          )
        : const SizedBox.shrink();
  }
}

File: native_upgrade/app/src/main/java/com/example/myapplication/ui/components/DeleteBirdDialog.kt
package com.example.app.ui.components

import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.example.app.birds.domain.model.Bird

@Composable
fun DeleteBirdDialog(
    showDialog: Boolean,
    birdToDelete: Bird?,
    onDeleteConfirmed: (Int) -> Unit,
    onDismiss: () -> Unit
) {
    if (showDialog && birdToDelete != null) {
        AlertDialog(
            onDismissRequest = onDismiss,
            title = { Text("Confirm Deletion") },
            text = { Text("Are you sure you want to delete ${birdToDelete.name}?") },
            confirmButton = {
                Button(
                    onClick = {
                        onDeleteConfirmed(birdToDelete.id)
                        onDismiss()
                    }
                ) {
                    Text("Delete")
                }
            },
            dismissButton = {
                Button(
                    onClick = onDismiss
                ) {
                    Text("Cancel")
                }
            }
        )
    }
}
